packages/aws-cdk/toolkit/toolkit.ts

```diff
@@ -0,0 +1,106 @@
+import * as chalk from 'chalk';
+
+/**
+ * Basic message structure for toolkit notifications.
+ * Messages are emitted by the toolkit and handled by the IoHost.
+ */
+interface IoMessage {
+  time: Date;
+  level: IoMessageLevel;
+  action: IoAction;
+  code: string;
+  message: string;
+  // Specify Chalk style for stdout/stderr, if TTY is enabled
+  style?: ((str: string) => string);
```

Comment:
This is a good thought. Is this currently used anywhere?

Beyond that, that - there's an interesting question for you: We currently use `chalk` inline in a number of places. How are we going to deal with that?

Might be easier to always chalk and use a [global setting](https://github.com/chalk/chalk?tab=readme-ov-file#chalklevel) to deal with color vs no color.

Reply:
Not currently used anywhere, but I thought this would be a nice thing to add for customers when we create the IoHost interface

Reply:
Actually I did some more recon on this and this is a very real problem I'm gonna need to address in a couple spots. This is enough though that I want to break that out into a separate pr, so I'll handle that in a follow up

Reply:
can you explain the problem here?